cfpb/cfpb-chart-builder
# Upgrade to highcharts 6

Closes:
- [GHE]/CFGOV/platform/issues/2963
- [GHE]/CFGOV/platform/issues/3095
- [GHE]/CFGOV/platform/issues/3091
- [GHE]/CFGOV/platform/issues/3071
- [GHE]/CFGOV/platform/issues/3027
- [GHE]/CFGOV/platform/issues/3024
- [GHE]/CFGOV/platform/issues/2991
- https://github.com/cfpb/cfpb-chart-builder/issues/71

## Changes

- Fixes issue with TileMap where tiles at the edge of the map had their borders clipped.

- Fixes issue with TileMap where cursor and border highlight was not consistent when rolling over tile square and tile text. Sets cursor for both interactions to be default pointer and a consistent 2px border.

- Fixes an issue where the tooltip legend line that is 1px high disappeared sometimes (presumably because it wasn't landing on a full pixel value).

- Normalizes tooltip appearance to not have a drop-shadow on the text and to be of "square" style across all three line charts (line, line-comparison, and line-index) and GeoMap, and "callout" style in bar chart.

- Simplifies responsive styles to two states (except for TileMap, which has it's own responsive nuances that should be re-visited in the future). This fixes an issue where the range selector buttons had several breakpoints, yet still didn't have correct horizontal alignment at several screen sizes.

- Moves chart-type specific styles to their own stylesheets, to make it easier to see the scope of the styles.

- Adds additional code comments and organization of what styles control what.

- Slight tweaks to the layout based on assumptions of what looks best.

- Removes navigator scrollbar, which does not appear necessary since the navigator can be dragged and it does nothing at the default appearance.


## Testing

- Run `./setup.sh && gulp build && gulp watch`
- Visit http://localhost:5000/ and resize the page to see responsive sizes. Check in Chrome, FF, IE10+, Edge, iOS simulator.
- Interact with charts by hovering and checking applicable interactive bits.

### Test on cfgov-refresh
 - Pull down this branch.
 - Check out https://github.com/cfpb/cfgov-refresh/tree/upgrade_babel branch on cfgov-refresh
 - Run `./frontend.sh`
 - Replace `cfgov-refresh/node_modules/cfpb-chart-builder` with `cfpb-chart-builder` base directory from this branch.
 - Run `gulp clean && gulp build`
 - Visit CCT pages and mortgage performance trends on localhost.

## Screenshots

Before: 

 - [Desktop](https://user-images.githubusercontent.com/704760/47681564-3083e280-dba0-11e8-8cd9-554e930da7ac.png)

 - [Mobile](https://user-images.githubusercontent.com/704760/47681573-38438700-dba0-11e8-9752-dacdc234b921.png)

After:

 - [Desktop](https://user-images.githubusercontent.com/704760/47681584-3ed1fe80-dba0-11e8-81ff-f6badc432a10.png)

 - [Mobile](https://user-images.githubusercontent.com/704760/47682109-ca985a80-dba1-11e8-8550-559de0c01473.png)

## Notes

- ~~**GeoMaps:** I'm unclear on how the GeoMap is set up on the live site, since chart builder contains a large number of custom styles in `demo.less` for the demo GeoMap. The version on the master branch and in this PR differ in coloring, but I'm unclear if this is just something to do with the demo or if this will break the live maps. We may need to merge this, make a release, and then vigorously test with the live site, and release patches if needed before finally merging the bumped version into cfgov-refresh.~~ These have been removed

- ~~**Firefox** does not show a thicker border on roll over on the GeoMap, unlike Chrome, which does show such a border. This is an existing issue.~~ Fixed in https://github.com/cfpb/cfpb-chart-builder/pull/152/commits/73a209e0800f5618f3687cf21e5b4fbc9e76c83b

- **IE** does not support `transform` on SVG elements, which was needed for some fine tuning. Consequently the range selector text is not vertically aligned and the GeoMap zoom button text isn't centered. I couldn't figure out a way around this, and it's an improvement over what's live. The tooltip is buggy in IE11, which we need to confirm on the live maps with this update to see if it's an issue of the demo or the live maps. It's an existing issue in the master branch.
